lib/model.js

```diff
@@ -1612,6 +1613,15 @@ module.exports = (function() {
 
     options.type = QueryTypes.BULKUPDATE;
 
+    // Remove values that are not in the options.fields
+    if (options.fields && options.fields instanceof Array) {
+      Object.keys(values).forEach(function(key) {
+        if (options.fields.indexOf(key) < 0) {
+          delete values[key];
```

Comment:
Correct, an `undefined` value is still a defined property.
This is not really a case we need to optimize so it's probably ok :)